### PR TITLE
Add CRT simulation to InTimes, Dirt, CRT-filtered samples

### LINKED ISSUE #212: Add CRT simulation to InTime and Dirt Workflow
Now that we have the CRT simulation working with the new LArG4 (PR #207), we need to make sure that:

- [x] we run it also in the in-time workflow
- [x] we run it also in the dirt workflow
- [x] we can make CRT-filtered samples

### PR DESCRIPTION
**Pull request by marcodeltutto**
This PR adds CRT simulation to InTimes, Dirt and CRT-filtered gen and g4 stages

Needs:
- `sbndcode` PR #207 (and PRs therein)
- `larsim` PR LArSoft/larsim#81

Fixes #212.

### CODE REVIEW

**Comment by wesketchum** [comment]
@marcodeltutto Realize this requires another PR to be in, but any reason that this can't be reviewed/made as ready to go as possible? If so, let's move from draft to full PR and request a review.

**Comment by marcodeltutto** [comment]
Hi @wesketchum, it's because I haven't finished testing. I'll mark it as ready-for-review soon

**Comment by pgreen135** [comment]
@marcodeltutto looks like there are some conflicts with develop, could you resolve these?

**Review by etyley — APPROVED** [review]
The fcls look good to me, minus some small questions about the intime cosmic workflow

File: sbndcode/JobConfigurations/standard/g4/g4_simphotontime_filter.fcl [hunk 1]
```diff
@@ -10,20 +10,24 @@
...
+physics.producers.genericcrtouttime: @local::sbnd_genericCRT
 
 # Set the appropriate input labels, to run geant4 only on the outtime cosmics
 physics.producers.simdriftintime.SimulationLabel: "ionandscintintime:priorSCE"
+physics.producers.genericcrtintime.LArG4Label: "larg4intime"
 physics.producers.larg4outtime.inputCollections: [ "GenInTimeSorter:outtime" ]
 physics.producers.ionandscintouttime.InputModuleLabels: ["larg4outtime"]
 physics.producers.pdfastsimouttime.SimulationLabel: "ionandscintouttime:priorSCE"
 physics.producers.simdriftouttime.SimulationLabel: "ionandscintouttime:priorSCE"
+physics.producers.genericcrtouttime.LArG4Label: "larg4outtime"
```
[the remark this answers is not in the document]
Comment (marcodeltutto) [reply sbndcode/JobConfigurations/standard/g4/g4_simphotontime_filter.fcl:30]: Thanks for spotting this! I added it with commit b1f35cb and checked we have `AuxDetSimChannels` in the end.

File: sbndcode/JobConfigurations/standard/standard_g4_sbnd.fcl [hunk 2]
```diff
@@ -123,6 +130,8 @@ outputs:
                     , "drop sim::SimEnergyDeposits_largeant_*_*"
                     # Drop the IonAndScint w/ SCE offsets applied
                     , "drop *_ionandscint__*"
+                    # Drop LArG4 AuxDetHits, now replaced by AuxDetSimChannels
```
[the remark this answers is not in the document]
Comment (marcodeltutto) [reply sbndcode/JobConfigurations/standard/standard_g4_sbnd.fcl:133]: Yes! Added it with commit b1f35cb.

**Comment by pgreen135** [comment]
@marcodeltutto https://github.com/LArSoft/larsim/pull/81 was not included in this weeks larsoft release, so as it currently stands I cannot include this into the production release

**Comment by marcodeltutto** [comment]
> @marcodeltutto [LArSoft/larsim#81](https://github.com/LArSoft/larsim/pull/81) was not included in this weeks larsoft release, so as it currently stands I cannot include this into the production release -- what do we want to do about this?

@pgreen135 We asked the larsoft team and they are working on it, we should have a larsoft patch release soon.

> could you also fix the conflict with develop

All fixed, also added a couple more commits since I had the chance ;)

**Review by jzennamo — APPROVED** [review]

**Comment by pgreen135** [comment]
trigger build

**Comment by pgreen135** [comment]
trigger build

**Comment by pgreen135** [comment]
trigger build

**Pull request merged**